### PR TITLE
Writing pacman-mirrors.conf

### PR DESCRIPTION
**Pull request by fhdk**
@guinux 
I have spotted some minor inconsistensies in writing **pacman-mirrors.conf**
The previous standard was to insert spaces around **`x = y`**.
Not having those affects pacman-mirrors in certain functions the app is checking the config file if a change should be written to a specific line. It has caused the change to be written to the end of the file.

    if "OnlyCountry = " in line then:

or

    if "Method = " in line then:

fails if it has been written as

    Method=

or

    OnlyCountry=

### CODE REVIEW

File: src/mirrors_config.vala [hunk 1]
```diff
@@ -116,16 +116,20 @@ namespace Pamac {
...
+									data.append ("Method = %s\n".printf (variant.get_string ()));
+								}
 							} else {
 								data.append (line + "\n");
 							}
 						} else if (line.contains ("OnlyCountry")) {
 							if (new_conf.lookup_extended ("OnlyCountry", null, out variant)) {
 								if (variant.get_string () == "ALL") {
-									data.append ("#%s\n".printf (line));
+									data.append ("# OnlyCountry = \n".printf (line));
```
Comment (philmmanjaro) [inline src/mirrors_config.vala]: This makes not so much sense to me. If **ALL** is selected shouldn't it be not **%s** as it was?

Comment (fhdk) [inline src/mirrors_config.vala]: That might have something to do with me having absolutely no idea of what %s contains when it is used. I was reading the code to check if pamac wrote the info with spaces as is expected by pacman-mirrors.

I simply assumed that %s only contained the values of OnlyCountry.

If it contains `OnlyCountry = ` then your are probably right. and then only %s should be there.

Comment (philmmanjaro) [inline src/mirrors_config.vala]: Maybe, however your change makes also sense. Let's see what @guinux will say.

Comment (philmmanjaro) [inline src/mirrors_config.vala]: This ends in:

```
mirrors_config.vala:120.23-120.51: error: Invalid type for argument 1
									data.append ("# Method = \n".printf (line));
									             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mirrors_config.vala:130.23-130.56: error: Too many arguments for specified format
									data.append ("# OnlyCountry = \n".printf (line));
									             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
mirrors_config.vala:130.23-130.56: error: Invalid type for argument 1
```

The **%s** would be the current **line**. When you have an issue with `"OnlyCountry=%s\n"` and the fix would be `"OnlyCountry = %s\n"`. So you have to adopt similar from the second case to the fist case and actually test your code first.

Comment (fhdk) [inline src/mirrors_config.vala]: Sorry about that. Though it seems very weird to me.

    data.append ("OnlyCountry = %s\n".printf (variant.get_string ()));

This line will fail as well?

Actually I copied the second if clause from OnlyCountry to Method and and changed OnlyCountry to Method and added spaces. I didn't think about testing it - it seemed totally logic to me - copying a working piece of code.

```
								if (variant.get_string () == "ALL") {
									data.append ("# OnlyCountry = \n".printf (line));
								} else {
									data.append ("OnlyCountry = %s\n".printf (variant.get_string ()));								}
```

Comment (fhdk) [inline src/mirrors_config.vala]: I have been looking at the code and it struck me

The statement

     "OnlyCountry = %sf \n".printf  (variant.get_string ());

is equivalent to python

    "OnlyCountry = {}\n".format (some_function())

So I have changed

    data.append ("# OnlyCountry = \n".printf (line));

to

    data.append ("# OnlyCountry = \n");

Which removes compile errors regarding the previous format

**Comment by philmmanjaro** [comment]
It compiles now. However, does it result now in the wanted **pacman-mirrors.conf** file?

**Comment by fhdk** [comment]
When looking at the code - I would say yes.

However my compile fails with this error:
```
valac	-o libpamac.so \
		-X -fPIC \
		-X --shared \
		-H pamac.h \
		--vapi=pamac.vapi \
		--library=libpamac \
		-X -w --pkg=posix -X -DGETTEXT_PACKAGE="pamac" --target-glib=2.38 \
		--pkg=gtk+-3.0 --pkg=gdk-3.0 --pkg=vte-2.91 -X -lm \
		--gresources=../resources/pamac.transaction.gresource.xml \
		../resources/transaction_resources.c \
		common.vala pamac_config.vala \
		package.vala transaction.vala history_dialog.vala choose_provider_dialog.vala transaction_sum_dialog.vala progress_box.vala \
		preferences_dialog.vala choose_ignorepkgs_dialog.vala mirrors_config.vala
preferences_dialog.vala:276.34-276.37: error: Argument 1: Cannot pass value to reference or output parameter
				ignorepkgs_liststore.remove (iter);
				                             ^^^^
Compilation failed: 1 error(s), 0 warning(s)
make[1]: *** [Makefile:94: libpamac.so] Error 1
make[1]: Leaving directory '/media/sources/manjaro/pamac/src'
make: *** [Makefile:14: all] Error 2
```
And I have no clue

File: src/mirrors_config.vala (hunk 1 above)
Comment (guinux) [inline src/mirrors_config.vala]: Hi,
I think just adding the missing spaces in "Method=%s\n" and "OnlyCountry=%s\n" is enough. Can you test it?

**Comment by philmmanjaro** [comment]
@fhdk: please see also https://github.com/manjaro/pamac/issues/235 and check with @guinux suggestion only adding the spaces as he explained.

**Comment by philmmanjaro** [comment]
I think 61f04da9a2372320fd22a75e9bc1e316d166ceb5 will make this one obsolete.

**Comment by fhdk** [comment]
I seems so - will pull the new code later and check again.

Thanks for looking at it :+1:

**Pull request closed without merging**